src/vm/values/types.nim

```diff
@@ -234,13 +240,63 @@ type
 {.hint: "'Value's inner type is currently '" & $sizeof(ValueObj) & "'.".}
 {.hints: off.}
 
-when sizeof(ValueObj) > 64: # At time of writing it was '56', 8 - 64 bit integers seems like a good warning site? Can always go smaller
+when sizeof(ValueObj) > 72: # At time of writing it was '72', 8 - 64 bit integers seems like a good warning site? Can always go smaller
     {.warning: "'Value's inner object is large which will impact performance".}
 
-template makeFuncAccessor*(name: untyped) =
-  template name*(val: Value): typeof(val.funcType.name) =
-    assert val.kind == Function
-    val.funcType.name
+proc readonly*(val: Value): bool {.inline.} = isReadOnly in val.flags
+proc `readonly=`*(val: Value, newVal: bool) {.inline.} = val.flags[isReadOnly] = newVal
+
+proc dirty*(val: Value): bool {.inline.} = isDirty in val.flags
+proc `dirty=`*(val: Value, newVal: bool) {.inline.} = val.flags[isDirty] = newVal
+
+proc dynamic*(val: Value): bool {.inline.} = isDynamic in val.flags
+proc `dynamic=`*(val: Value, newVal: bool) {.inline.} = val.flags[isDynamic] = newVal
+
+proc b*(val: Value): VLogical {.inline.} =
+    assert val.kind == Logical
+    assert (val.flags * {isMaybe, isTrue}).len <= 1 # Ensure we do not have {isMaybe, isTrue}
+    if val.flags * {isMaybe} == {isMaybe}:
+        Maybe
+    elif val.flags * {isTrue} == {isTrue}:
+        True
+    else:
+        False
```

Comment:
```suggestion
    if isTrue in val.flags: True
    elif isFalse in val.flags: False
    else: Maybe
```

Could we add another `isFalse` flag? I think it would many things simpler (since a) we know by definition that only Logical values can have them, b) If a Value is a Logical one, one of the three *will* exist no matter what). 

Also it would help create helpers like `template IsFalse*(v: Value):untyped = isFalse in v.flags` for fast checks (to replace things like `x.b == True` which look like a total overkill this way, but are pretty much everywhere...)